From 4f981f6ab614a908f912f7dc147b248f96b498a2 Mon Sep 17 00:00:00 2001
From: satmandu <satadru@umich.edu>
Date: Wed, 26 Dec 2018 12:36:28 -0500
Subject: [PATCH] Additional fixes for current_kernel_time in 4.20 on 0.7.x

While https://github.com/zfsonlinux/spl/commit/cd1b28e0cfa923db988611df28958f7b84b53530 fixed the 4.20 current_kernel_time issues in spl as per https://github.com/zfsonlinux/zfs/pull/8074 in .07.x, current_kernel_time is still being used inside zpios-ctl.h in 0.7.x.

I believe this may fix this issue in 0.7.x to allow it to compile successfully for kernel 4.2.0.
---
 include/zpios-ctl.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/include/zpios-ctl.h b/include/zpios-ctl.h
index f17f1153260..068c9eaf173 100644
--- a/include/zpios-ctl.h
+++ b/include/zpios-ctl.h
@@ -181,9 +181,12 @@ zpios_timespec_t
 zpios_timespec_now(void)
 {
 	zpios_timespec_t zts_now;
-	struct timespec ts_now;
+	#if defined(HAVE_INODE_TIMESPEC64_TIMES)
+		inode_timespec_t ts_now = current_kernel_time64();
+	#else
+		inode_timespec_t ts_now = current_kernel_time();
+	#endif
 
-	ts_now = current_kernel_time();
 	zts_now.ts_sec  = ts_now.tv_sec;
 	zts_now.ts_nsec = ts_now.tv_nsec;
 
